ibmcloudprovider/volume_provider.go

```diff
@@ -0,0 +1,160 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package ibmcloudprovider ...
+package ibmcloudprovider
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/IBM/ibm-csi-common/pkg/utils"
+	"github.com/IBM/ibmcloud-volume-file-vpc/common/registry"
+	provider_util "github.com/IBM/ibmcloud-volume-file-vpc/file/utils"
+	vpcconfig "github.com/IBM/ibmcloud-volume-file-vpc/file/vpcconfig"
+	"github.com/IBM/ibmcloud-volume-interface/config"
+	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	"github.com/IBM/ibmcloud-volume-interface/provider/local"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+)
+
+// IBMCloudStorageProvider Provider
+type IBMCloudStorageProvider struct {
+	ProviderName   string
+	ProviderConfig *config.Config
+	Registry       registry.Providers
+	ClusterInfo    *utils.ClusterInfo
+}
+
+var _ CloudProviderInterface = &IBMCloudStorageProvider{}
+
+// NewIBMCloudStorageProvider ...
+func NewIBMCloudStorageProvider(configPath string, logger *zap.Logger) (*IBMCloudStorageProvider, error) {
+	logger.Info("NewIBMCloudStorageProvider-Reading provider configuration...")
+	// Load config file
+	conf, err := config.ReadConfig(configPath, logger)
+	if err != nil {
+		logger.Error("Error loading configuration")
+		return nil, err
+	}
+	// Get only VPC_API_VERSION, in "2019-07-02T00:00:00.000Z" case vpc need only 2019-07-02"
+	dateTime, err := time.Parse(time.RFC3339, conf.VPC.APIVersion)
+	if err == nil {
+		conf.VPC.APIVersion = fmt.Sprintf("%d-%02d-%02d", dateTime.Year(), dateTime.Month(), dateTime.Day())
+	} else {
+		logger.Warn("Failed to parse VPC_API_VERSION, setting default value")
+		conf.VPC.APIVersion = "2021-04-20" // setting default values
+	}
+
+	logger.Info("Fetching clusterInfo")
+	clusterInfo, err := utils.NewClusterInfo(logger)
+	if err != nil {
+		logger.Error("Unable to load ClusterInfo", local.ZapError(err))
+		return nil, err
+	}
+	logger.Info("Fetched clusterInfo..")
+	if conf.Bluemix.Encryption || conf.VPC.Encryption {
+		if os.Getenv("IKS_ENABLED") == "True" {
+			// api Key if encryption is enabled
+			logger.Info("Creating NewAPIKeyImpl...")
+			apiKeyImp, err := utils.NewAPIKeyImpl(logger)
+			if err != nil {
+				logger.Error("Unable to create API key getter", local.ZapError(err))
+				return nil, err
+			}
+			logger.Info("Created NewAPIKeyImpl...")
+			err = apiKeyImp.UpdateIAMKeys(conf)
+			if err != nil {
+				logger.Error("Unable to get API key", local.ZapError(err))
+				return nil, err
+			}
+		}
+	}
+
+	// Update the CSRF  Token
+	if conf.Bluemix.PrivateAPIRoute != "" {
```

Comment:
Now its clear that API settings not required for file, can be remove these settings